rfcs/45-stageless.md

```diff
@@ -0,0 +1,557 @@
+# Feature Name: `stageless`
+
+## Summary
+
+The existing stage boundaries result in a large number of complications: preventing us from operating across stages.
+Moreover, several related features (run criteria, states and fixed timesteps) are rather fragile and overly complex.
+This tangled mess needs to be holistically refactored: simplifying the scheduling model down into a single `Schedule`, moving towards an intent-based configuration style, taking full advantage of labels and leveraging exclusive systems to handle complex logic.
+
+## Motivation
+
+The fundamental challenges with the current stage-driven scheduling abstraction are [numerous](https://github.com/bevyengine/bevy/discussions/2801).
+Of particular note:
+
+- Dependencies do not work across stages.
+- States do not work across stages.
+- Plugins can add new, standalone stages.
+- Run criteria (including states!) cannot be composed.
+- The stack-driven state model is overly elaborate and does not enable enough use cases to warrant its complexity.
+- Fixed timestep run criteria do not correctly handle the logic needed for robust game physics.
+- The architecture for turn-based games and other even slightly unusual patterns is not obvious.
+- We cannot easily implement a builder-style strategy for configuring systems.
+
+Unfortunately, all of these problems are deeply interwoven. Despite our best efforts, fixing them incrementally at either the design or implementation stage is impossible, as it results in myopic architecture choices and terrible technical debt.
+
+## User-facing explanation
+
+This explanation is, in effect, user-facing documentation for the new design.
+In addition to a few new concepts, it throws out much of the current system scheduling design that you may be familiar with.
+The following elements are substantially reworked:
+
+- schedules (flattened)
+- run criteria (can no longer loop, are now systems)
+- states (simplified, no longer purely run-criteria powered)
+- fixed time steps (no longer a run criteria)
+- exclusive systems (no longer special-cased)
+- command processing (now performed in a `flush_commands` exclusive system)
+- labels (can now be directly configured)
+- system sets (now just used to mass-apply a label to a group of systems)
+- stages (yeet)
+
+### Scheduling overview and intent-based configuration
+
+Systems in Bevy are stored in a `Schedule`: a collection of **configured systems**.
+Each frame, the `App`'s `runner` function will run the schedule,
+causing the **scheduler** to run the systems in parallel using a strategy that respects all of the configured constraints (or panic, if it is **unsatisfiable**).

```

Comment:
> **unsatisfiable**

If at all possible, this should link to the fragment of the document explaining this term. 

Afaik, GitHub Flavored Markdown allows enough embedded html to create a `<span>` with an ID; if not, a header would do, even if unsatisfying. (Note that at this point, I haven't read all of this document, so I don't even know if there is an explanation to be found)

Reply:
Added a better explanation here.